docs/proposals/volume-provisioning.md

```diff
@@ -0,0 +1,315 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+Real Kubernetes clusters have a variety of volumes which differ widely in
+size, iops performance, retention policy, and other characteristics.
+Administrators need a way to dynamically provision volumes of these different
+types to automatically meet user demand.
+
+A new mechanism called 'storage classes' is proposed to provide this
+capability.  Storage classes build upon the label selection feature added in
+Kubernetes 1.3.
+
+## Motivation
+
+In Kubernetes 1.2, an alpha form of limited dynamic provisioning was added
+that allows a single volume type to be provisioned in clouds that offer
+special volume types.
+
+In Kubernetes 1.3, a label selector was added to persistent volume claims to
+allow administrators to create a taxonomy of volumes based on the
+characteristics important to them, and to allow users to make claims on those
+volumes based on those characteristics.  This allows flexibility when claiming
+existing volumes; the same flexibility is needed when dynamically provisioning
+volumes.
+
+After gaining experience with dynamic provisioning after the 1.2 release, we
+want to create a more flexible feature that allows configuration of how
+different storage classes are provisioned and supports provisioning multiple
+types of volumes within a single cloud.
+
+## Constraints and Assumptions
+
+The proposed design should:
+
+1.  Build upon the label selection capabilities added in Kubernetes 1.3
+2.  Not interfere with existing semantics of manually-created volumes
+3.  Not require users to know or understand the differences between
+    volumes (ie, Kubernetes should not dictate any particular set of
+    characteristics to administrators to think in terms of)
+
+This proposal will **not** deal with out-of-tree provisioners, but we should
+understand that they are desired by the community, and so we should be careful
+not to make them impossible to implement in the future.
+
+## Use Cases
+
+1.  As an administrator, I want to be able to provision multiple configurations
```

Comment:
My template is to use a list.  It has a single item in it for now.  I try not to churn on formatting stuff like that since you will spend a lot of time rewriting prose that doesn't add value as your proposal changes.